ext/standard/var.c

```diff
@@ -36,13 +36,28 @@ struct php_serialize_data {
 
 #define COMMON (is_ref ? "&" : "")
 
+static void php_print_escaped(zend_string *str) {
+	/* Escape null bytes in strings.
+	 * This does not use php_addcslashes(), because that function formats null bytes as \000. */
+	const char *p = ZSTR_VAL(str);
+	const char *e = p + ZSTR_LEN(str);
+	while (p < e) {
+		const char *p2 = p;
+		while (p2 < e && *p2 != '\0') p2++;
+		if (p != p2) PHPWRITE(p, p2 - p);
+		if (p2 == e) break;
+		PHPWRITE("\\0", 2);
```

Comment:
Writing a longer thread on internals@, but I think only escaping this one character is going to cause more harm than good.